### PR TITLE
Remove /run mount in fstab to prevent auto remounts

### PR DESCRIPTION
**Pull request by Squoimote**
Auto remounts of `/run` when executing programs cause corruptions.

Material: PiKVM 4 Plus (probably same with others)

The issue was:
- Broken systemd, saying not executed at the boot and not PID 1 (even it is)
- Broken Janus, who progressively use 100% of the CPU (after 5 minutes and more)
- Hard rebooting in this condition can break the boot sequence
- Probably other issues...

Steps to reproduce:
Flash the image on sd card
Go to the terminal in root
Make writable with `rw`
Edit `/etc/fstab` to add an USB stick mount point (maybe optional)
Run `mount -a`
Try to use systemd, `cat /etc/resolv.conf`, `ping google.com` or monitor janus...

image
image
image

### CODE REVIEW

**Comment by mdevaev** [comment]
Hello. `noauto` will break os on booting since `/run` will not be mounted at all.

**Comment by Squoimote** [comment]
Hello, nope, mine is booting very well :)

image

**Comment by Squoimote** [comment]
Hello (again ^^), I found why it is working in this case by reading documentation.
It needs more troubleshoot of what is calling the partition and analyze when it is mount during the boot sequence to ensure it mounts before starting programs.


From what I know:

Mount man says:
```
       -a, --all
           Mount all filesystems (of the given types) mentioned in fstab
           (except for those whose line contains the noauto keyword).
           The filesystems are mounted following their order in fstab.
```
```
       noauto
           Can only be mounted explicitly (i.e., the -a option will not
           cause the filesystem to be mounted).
```
See: https://linux.die.net/man/8/mount


But in fact, some have issues using `noauto`:
https://askubuntu.com/questions/439719/prevent-mounting-a-partition-in-fstab
https://askubuntu.com/questions/421585/how-can-i-prevent-auto-mounting-of-a-partition-in-fstab

The systemd.mount man page adds some explanations of what happen:
https://man.archlinux.org/man/systemd.mount.5
```
noauto, auto
	With noauto, the mount unit will not be added as a dependency for local-fs.target or remote-fs.target.
	This means that it will not be mounted automatically during boot, unless it is pulled in by some other unit.
	The auto option has the opposite meaning and is the default. 
```

So, in this case, it mounts `/run` during the boot when I write `noauto` in `/etc/fstab`.
I just need to find what systemd process is calling `/run` and see when, during the boot sequence, `/run` mount.

**Comment by mdevaev** [comment]
Okay.

**Comment by Squoimote** [comment]
Hello (for the last time i hope ^^)

I was questioning myself about how preventing the remount of `/run` in `/etc/fstab` before knowing what `/run` is.
Explanation of `/run` tmpfs:
https://unix.stackexchange.com/questions/13972/what-is-this-new-run-filesystem
http://www.h-online.com/open/news/item/Linux-distributions-to-include-run-directory-1219006.html

So `/run` is mounted early in the boot process and is not handle by the fstab generator.
As a proof, `systemctl cat run.mount` will return an error (`No files found for run.mount.`) even `/run` is in the `/etc/fstab`.
Knowing this, why `/run` mount is in `/etc/fstab`?


We can test it, with and without!

With `/run` mount point in `/etc/fstab` (configured as default):
```
[root@pikvm ~]# cat /proc/mounts | grep ^run
run /run tmpfs rw,nosuid,nodev,relatime,mode=755 0 0
```
```
[root@pikvm ~]# df -h
Filesystem      Size  Used Avail Use% Mounted on
dev             838M     0  838M   0% /dev
run             908M   11M  898M   2% /run
/dev/mmcblk0p3  5.9G  2.5G  3.1G  44% /
tmpfs           908M  124K  908M   1% /dev/shm
tmpfs           908M     0  908M   0% /tmp
tmpfs           908M     0  908M   0% /var/lib/dhclient
tmpfs           908M     0  908M   0% /var/lib/dhcpcd
tmpfs           908M     0  908M   0% /var/lib/private
tmpfs           908M  4.0K  908M   1% /var/lib/systemd
tmpfs           908M  4.0K  908M   1% /var/log
tmpfs           908M     0  908M   0% /var/tmp
/dev/mmcblk0p1  256M   43M  214M  17% /boot
/dev/mmcblk0p2  230M   79K  226M   1% /var/lib/kvmd/pst
/dev/mmcblk0p4   23G  2.1M   23G   1% /var/lib/kvmd/msd
```

Without `/run` mount point in `/etc/fstab`:
```
[root@pikvm ~]# cat /proc/mounts | grep ^run
run /run tmpfs rw,nosuid,nodev,relatime,mode=755 0 0
```
```
[root@pikvm ~]# df -h
Filesystem      Size  Used Avail Use% Mounted on
dev             838M     0  838M   0% /dev
run             908M  8.5M  900M   1% /run
/dev/mmcblk0p3  5.9G  2.5G  3.1G  44% /
tmpfs           908M  124K  908M   1% /dev/shm
tmpfs           908M     0  908M   0% /tmp
tmpfs           908M     0  908M   0% /var/lib/dhclient
tmpfs           908M     0  908M   0% /var/lib/dhcpcd
tmpfs           908M     0  908M   0% /var/lib/private
tmpfs           908M  4.0K  908M   1% /var/lib/systemd
tmpfs           908M  4.0K  908M   1% /var/log
tmpfs           908M     0  908M   0% /var/tmp
/dev/mmcblk0p1  256M   43M  214M  17% /boot
/dev/mmcblk0p2  230M   79K  226M   1% /var/lib/kvmd/pst
/dev/mmcblk0p4   23G  2.1M   23G   1% /var/lib/kvmd/msd
```

In conclusion, you can safely remove `/run` from `/etc/fstab` without any impact.
The `/run` mount point is a special mount point directly managed by systemd.
Removing this line will prevent future people to break their fully configured system by fixing the issue I have with `mount -a`.
Because it made me really mad. 👎

If you do not trust me, test it ;)
You are great, best wishes 🥇

**Comment by mdevaev** [comment]
I believe you and it looks like you're right. But I would like to find a description of this mechanism in the systemd documentation, and so far I have not succeeded. Maybe I'm not looking well :/

**Comment by Squoimote** [comment]
Well, I'll ask a previous coworker about this.
I lost my Systemd learning, he can send me this back...
In parallel, i'll search in Systemd documentation or in FHS for informations.

**Comment by mdevaev** [comment]
My friend systemd-guru confirmed this. Thanks again.
